tv_apps/smart-home/test/unit/home_test.js

```diff
@@ -0,0 +1,382 @@
+        assert.isTrue(spy.calledOnce);
+      });
+
+      test('should reset filter', function() {
+        var spy = this.sinon.spy(subject.filterManager, 'resetFilter');
+        subject.onVisibilityChange();
+        assert.isTrue(spy.calledOnce);
+      });
+    });
+  });
+
+  suite('initClock >', function() {
+    test('should set time after time elapsed', function() {
+      subject.initClock();
+      var stub = this.sinon.stub(subject, 'updateClock');
+      fakeTimer.tick(1000);
+      assert.isTrue(stub.calledOnce);
+    });
+  });
+
+  suite('onCardInserted >', function() {
+    test('should insert a card into scrollable', function() {
+      var card1 = new Folder({});
+      var card2 = new Folder({});
+      var stub = this.sinon.stub();
+      var testCase1 = stub.withArgs(sinon.match.truthy, 0);
+      var testCase2 = stub.withArgs(sinon.match.truthy, 1);
+      subject.cardScrollable.insertNodeBefore = stub;
+
+      subject.onCardInserted(subject.cardScrollable, card1, 0);
+      assert.isTrue(testCase1.calledOnce);
+      assert.isFalse(testCase2.calledOnce);
+      subject.onCardInserted(subject.cardScrollable, card2, 1);
+      assert.isTrue(testCase2.calledOnce);
+    });
+  });
+
+  suite('onCardRemoved >', function() {
+    test('should remove specified cards from scrollable', function() {
+      var stub = this.sinon.stub();
+      subject.cardScrollable.removeNodes = stub;
+      subject.onCardRemoved(subject.cardScrollable, [0, 1]);
+      assert.isTrue(stub.withArgs([0, 1]).calledOnce);
+    });
+  });
+
+  suite('onMove >', function() {
+    test('should ask spatial navigator for next focus on non-scrollable target',
+    function() {
+      subject.spatialNavigator.m_focusedElement = document.createElement('div');
+      var stub = this.sinon.stub();
+      subject.spatialNavigator.move = stub;
+      subject.onMove('left');
+      assert.isTrue(stub.calledWith('left'));
+    });
+
+    test('should ask scrollable to move its focus on scrollable target',
+    function() {
+      var scrollable = new XScrollable();
+      scrollable.move = this.sinon.stub();
+      scrollable.move.returns(true);
+      subject.spatialNavigator.m_focusedElement = scrollable;
+      subject.onMove('right');
+      assert.isTrue(scrollable.move.calledWith('right'));
+    });
+
+    test('should ask scrollable to move its focus on scrollable target, but ' +
+    'fall-back to spatial navigator when scrollable didn\'t handle it',
+    function() {
+      var scrollable = new XScrollable();
+      scrollable.move = this.sinon.stub();
+      scrollable.move.returns(false);
+      subject.spatialNavigator.m_focusedElement = scrollable;
+      var spatialNavigatorMoveStub = this.sinon.stub();
+      subject.spatialNavigator.move = spatialNavigatorMoveStub;
+      subject.onMove('up');
+      assert.isTrue(scrollable.move.calledWith('up'));
+      assert.isTrue(spatialNavigatorMoveStub.calledWith('up'));
+    });
```

Comment:
I think we should also test `onMove` when we are in edit mode.
That is to say, to examine if we bypass logic of `onMove` or not when `this.edit.onMove()` returns true.
However, I can accept that we add this test in next patch of unit test.